### PR TITLE
fix: polish examples - peer deps, env vars, endpoint defaults

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

Fixing some issues and minor discrepancies discovered while working on e2e tests. Namely:

- the examples had conflicting peer deps so `npm install` did not work on new versions,
- updated peer deps to the latest available ones.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Internal change (a change which is not visible to the package consumers)
- [ ] Documentation change or requires a documentation update

# How Has This Been Tested?

- [ ] Tested manually
- [x] Added automated tests

# Checklist:

- [ ] Unit tests have been added/updated
- [ ] Documentation has been updated
- [ ] Change file has been generated (`npm run change:new`)
- [ ] Delete this branch (after the PR is merged)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,33 +106,33 @@
...
+    "@opentelemetry/instrumentation-pino": "~0.25.0",
+    "@opentelemetry/instrumentation-redis": "~0.25.0",
+    "@opentelemetry/instrumentation-winston": "~0.25.0",
+    "opentelemetry-instrumentation-amqplib": "~0.25.0",
+    "opentelemetry-instrumentation-aws-sdk": "~0.25.0",
+    "opentelemetry-instrumentation-elasticsearch": "~0.25.0",
+    "opentelemetry-instrumentation-kafkajs": "~0.25.0",
+    "opentelemetry-instrumentation-mongoose": "~0.25.0",
+    "opentelemetry-instrumentation-sequelize": "~0.25.0",
+    "opentelemetry-instrumentation-typeorm": "~0.25.0"
```
Comment (seemk) [inline package.json:135]: In npm the 0.25.0 versions are tagged as next as opposed to latest (e.g. instrumentation-http), is this fine?

File: examples/express/package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
   },
   "dependencies": {
     "@opentelemetry/exporter-jaeger": "^0.24.0",
-    "@opentelemetry/instrumentation-express": "^0.24.0",
+    "@opentelemetry/instrumentation-express": "^0.23.0",
```
Comment (seemk) [inline examples/express/package.json:41]: Would it make sense to use `^0.25.0` here?

**Review by seemk — APPROVED** [review]
Some minor questions

File: examples/express/package.json (hunk 2 above)
Comment (Rauno56) [reply examples/express/package.json:41]: Can't. The released version of @splunk/otel@0.12 requires it to be `0.23.x`.

File: package.json (hunk 1 above)
Comment (Rauno56) [reply package.json:135]:
```shell
(test/e2e-tests)$ npm info @opentelemetry/instrumentation-knex dist-tags # same for most of the packages
{ latest: '0.25.0' }
(test/e2e-tests)$ npm info @opentelemetry/instrumentation-http dist-tags # same for http and grpc
{ latest: '0.24.0', canary: '0.25.1-alpha.24', next: '0.25.0' }
```

You are right in the context of `instrumentation-http` and `instrumentation-grpc`, and that's why I've left those to `~0.24.0` and only upgraded the others to `~0.25.0`.

**Comment by Rauno56** [comment]
Thanks for the quick review @seemk!

File: package.json (hunk 1 above)
Comment (seemk) [reply package.json:135]: Ah I missed the 0.24.0

**Pull request merged**